### PR TITLE
expand mtr coverage

### LINKED ISSUE #837: Audit/improve custom type tests coverage
Audit and increase custom type test coverage. Improve validation and developer-facing checks to make custom types easier to implement and to help authors catch incorrect or incomplete implementations early.

### PR DESCRIPTION
**Pull request by robgolebiowski-vsql**
Add mtr coverage for existing custom-type behavior plus the resolve_params sizing fix: invalid resolved persisted_length / max_decode_buffer_length, empty int_to_params, unknown/extra parameter keys, the mutating resolve_params param rewrite, intrinsic defaults (str/vdf/parameterized + error paths), and the absence of a VEF-level upper bound on declared sizes. Adds the edge-case and intrinsic-default test types to vsql-test-only.

Closes https://github.com/villagesql/villagesql-server/issues/837

### CODE REVIEW

File: villagesql/test-extensions/vsql-test-only/src/extension.cc [hunk 1]
```diff
@@ -685,21 +704,24 @@ constexpr auto PVEC =
...
-//                             than max_persisted_length (upper-bound check).
-//   BAD_LEN_PARAM_TYPE(N)   - fixed-length; resolve_params returns
-//                             persisted_length = -1 (must be > 0).
-//   VAR_SETS_PERSISTED(N)   - variable-length; resolve_params returns a
-//   positive
-//                             persisted_length (must stay <= 0).
-//   BAD_DECODE_BUF(N)       - resolve_params resolves a non-positive
-//                             max_decode_buffer_length (must be > 0).
+//   BAD_LEN_PARAM_TYPE(N)   - resolve_params returns persisted_length = -1 for
+//   a
```
Comment (villagestevers) [inline villagesql/test-extensions/vsql-test-only/src/extension.cc]: bad wrapping

File: mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Scenario: the mutating resolve_params overload can rewrite its parameter map,
+# and the server adopts the rewritten set as the canonical parameters. Here the
+# type removes (normalizes away) an unrecognized key.
+#
+# Fixture: vsql_test_only exposes DROP_PARAM_TYPE, whose (mutating) resolve_params
+# keeps 'length' and erases every other key.
+
+INSTALL EXTENSION vsql_test_only;
+
+--echo # An unrecognized key is dropped by resolve_params; the column is created
```
Comment (villagestevers) [inline mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test]: Should this be the case, or it should be an error - i think an error. Maybe we should discuss first.

Oh I guess this is kind of up to the implementer? Let's discuss.

File: mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result [hunk 3]
```diff
@@ -0,0 +1,14 @@
+INSTALL EXTENSION vsql_test_only;
+# A fixed type with an over-large persisted_length installs fine, but
+# CREATE TABLE is rejected by MySQL's column-length limit.
+CREATE TABLE t (id INT PRIMARY KEY, v HUGE_FIELD);
+ERROR 42000: Column length too big for column 'v' (max = 65535); use BLOB or TEXT instead
```
Comment (villagestevers) [inline mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result]: This is an odd message for us to print. Maybe these should be generic villagesql errors with a message explaining that the type is too large for what is supported today. We should open a feature (if we don't have it already) for BLOB support for custom types and refer to that issue so people can upvote it if they see it.

File: villagesql/test-extensions/vsql-test-only/src/extension.cc (hunk 1 above)
Comment (robgolebiowski-vsql) [inline villagesql/test-extensions/vsql-test-only/src/extension.cc]: fixed

File: mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result (hunk 3 above)
Comment (robgolebiowski-vsql) [inline mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result]: I have changed in this PR (the error message + added the mechanism for the max persisted_length/max_persisted_length values). I've opened a new feature request for blob support: https://github.com/villagesql/villagesql-server/issues/914

File: mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test (hunk 2 above)
Comment (robgolebiowski-vsql) [inline mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test]: As discussed. I've removed this test. I have created a new PR to pre-validate custom types parameters before they are passed to resolve_params.
Before calling resolve_params:
reject a duplicate name, a nameless parameter and a parameter with no value.
After calling resolve_params:
reject a rewrite that dropped a parameter.
I want to stack the new PR on top of this one. 
PR: https://github.com/villagesql/villagesql-server/pull/915

File: villagesql/test-extensions/vsql-test-only/src/extension.cc (hunk 1 above)
Comment (robgolebiowski-vsql) [reply villagesql/test-extensions/vsql-test-only/src/extension.cc]: done

File: mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result (hunk 3 above)
Comment (robgolebiowski-vsql) [reply mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result]: I have changed in this PR (the error message + added the mechanism for the max persisted_length/max_persisted_length values). I've opened a new feature request for blob support: https://github.com/villagesql/villagesql-server/issues/914

File: mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test (hunk 2 above)
Comment (robgolebiowski-vsql) [reply mysql-test/suite/villagesql/extension/t/extension_resolve_params_rewrite.test]: As discussed, fixed in a separate PR: https://github.com/villagesql/villagesql-server/pull/915

File: villagesql/test-extensions/vsql-test-only/src/extension.cc [hunk 4]
```diff
@@ -1134,60 +1103,38 @@ constexpr auto OVERSIZED_PARAM_TYPE =
...
-        .from_string<&param_test_encode>()
-        .to_string<&param_test_decode>()
-        .compare<&param_test_compare>()
-        .build();
-
-// ---- Over-large storage-size test types ---------------------------------
+// ---- Storage-size boundary test type -------------------------------------
+//
+// A custom column is backed by a VARBINARY field, so a type may declare at most
+// 65532 bytes of storage per value.
```
Comment (villagestevers) [inline villagesql/test-extensions/vsql-test-only/src/extension.cc:1109]: The other places mentioning this make it clear it is 65535 minus the 2 byte length and 1 byte null indicator. Should this also say that?

File: mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result (hunk 3 above)
Comment (villagestevers) [reply mysql-test/suite/villagesql/extension/r/extension_length_no_upper_bound.result]: I thought that would have been a duplicate but i guess not.

File: villagesql/veb/validate.cc [hunk 5]
```diff
@@ -98,6 +112,27 @@ std::optional<ValidatedRegistration> parse_extension_registration(
         error_out = "type '" + type_name + "' failed validation";
         return std::nullopt;
       }
+
+      // A fixed non-parameterized type declares persisted_length
+      // (max_persisted_length stays 0); parameterized and variable-length types
+      // declare max_persisted_length, and a parameterized type's resolved
+      // persisted_length is checked against it at DDL time, so bounding the
+      // declared values bounds every parameterization.
+      const int64_t declared_length =
```
Comment (villagestevers) [inline villagesql/veb/validate.cc:121]: I wonder why we check this explicitly. I am fine with it, but if there are other columns, wouldn't we realize eventually that the size is too large?

**Review by villagestevers — APPROVED** [review]

File: villagesql/veb/validate.cc (hunk 5 above)
Comment (robgolebiowski-vsql) [reply villagesql/veb/validate.cc:121]: Yes, but the error would be reported by MySQL server layer and the error is weird:ERROR 42000: Column length too big for column 'v' (max = 65535); use BLOB or TEXT instead - as previously it did.

File: villagesql/test-extensions/vsql-test-only/src/extension.cc (hunk 4 above)
Comment (robgolebiowski-vsql) [reply villagesql/test-extensions/vsql-test-only/src/extension.cc:1109]: changed
